### PR TITLE
Job Scheduling

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
see #31 

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual.
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Regression tests have to complete successfully.
- [ ] 5. If significant functionality is added, there must be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test.
- [ ] 6. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.

### CODE REVIEW

File: src/CapitalInvestments/PyomoModels/ModelBase.py [hunk 1]
```diff
@@ -5,49 +5,26 @@
   @author: wangc, mandd
 """
```
Comment (mandd) [inline src/CapitalInvestments/PyomoModels/ModelBase.py:6]: This changes are already present in PR #28, no?

File: src/CapitalInvestments/PyomoModels/PySPBase.py [hunk 2]
```diff
@@ -0,0 +1,374 @@
+# Copyright 2020, Battelle Energy Alliance, LLC
```
Comment (mandd) [inline src/CapitalInvestments/PyomoModels/PySPBase.py:1]: This changes are already present in PR #28, no?

File: src/CapitalInvestments/PyomoModels/RCPSP.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+      @ In, model, pyomo model instance, pyomo abstract model
+      @ Out, model, pyomo model instance, pyomo abstract model
+    """
+    model.obj = pyomo.Objective(rule=self.objExpression, sense=self.sense)
+    return model
+
+  ########### start constraint functions
+  @staticmethod
+  def constraintX(model, j):
+    return sum(model.x[j, t]  for t in model.time_periods) - 1 == 0
```
Comment (mandd) [inline src/CapitalInvestments/PyomoModels/RCPSP.py:201]: missing docstrings

File: src/CapitalInvestments/PyomoModels/RCPSP.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+  # Constraint on predecessors
+  @staticmethod
+  def predecessorsConstraint(model, jp, p, t):
```
Comment (mandd) [inline src/CapitalInvestments/PyomoModels/RCPSP.py:205]: missing docstrings

File: src/CapitalInvestments/PyomoModels/RCPSP.py [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    return sum(model.x[model.task_successors[(jp,p)], tprime] for tprime in model.time_periods if tprime <= t) <= sum(model.x[jp, tprime] for tprime in model.time_periods if tprime <= t - model.task_duration[model.task_successors[(jp,p)]])
+
+  #Constraint on resources
+  @staticmethod
+  def resourcesConstraint(model, r, t):
+    return (0, sum(sum(model.task_resource_consumption[j, r] * model.x[j, tprime] for tprime in model.time_periods if tprime >= t and tprime <= t + model.task_duration[j]-1) for j in model.tasks), model.available_resources[r])
```
Comment (mandd) [inline src/CapitalInvestments/PyomoModels/RCPSP.py:224]: missing docstrings

**Review by mandd — CHANGES_REQUESTED** [review]

File: src/CapitalInvestments/PyomoModels/ModelBase.py (hunk 1 above)
Comment (wangcj05) [reply src/CapitalInvestments/PyomoModels/ModelBase.py:6]: I'm sorry I forget to mention. This PR is based on PR#28. PR#28 should be merged first before this PR.

**Comment by wangcj05** [comment]
It seems we have some problem with LOGOS regression testing machine. I have tested this PR on my local machine, and all tests are green:

image

**Review by mandd — APPROVED** [review]

**Comment by mandd** [comment]
Tests are green on my machine as well. merging

**Pull request merged**